arangod/Agency/AgencyComm.cpp

```diff
@@ -1181,23 +1181,23 @@ AgencyCommResult AgencyComm::sendTransactionWithFailover(AgencyTransaction const
 bool AgencyComm::ensureStructureInitialized() {
   LOG_TOPIC(TRACE, Logger::AGENCYCOMM) << "checking if agency is initialized";
 
-  while (true) {
-    while (shouldInitializeStructure()) {
-      LOG_TOPIC(TRACE, Logger::AGENCYCOMM)
-          << "Agency is fresh. Needs initial structure.";
-      // mop: we are the chosen one .. great success
-
-      if (tryInitializeStructure()) {
-        LOG_TOPIC(TRACE, Logger::AGENCYCOMM)
-            << "Successfully initialized agency";
-        break;
-      }
+  auto serverFeature =
+    application_features::ApplicationServer::getFeature<ServerFeature>("Server");
+  while (!serverFeature->isStopping() && shouldInitializeStructure()) {
 
-      LOG_TOPIC(WARN, Logger::AGENCYCOMM)
-          << "Initializing agency failed. We'll try again soon";
-      // We should really have exclusive access, here, this is strange!
-      std::this_thread::sleep_for(std::chrono::seconds(1));
+    LOG_TOPIC(TRACE, Logger::AGENCYCOMM)
+      << "Agency is fresh. Needs initial structure.";
+    
+    if (tryInitializeStructure()) {
+      LOG_TOPIC(TRACE, Logger::AGENCYCOMM)
+        << "Successfully initialized agency";
+      break;
     }
+    
+    LOG_TOPIC(WARN, Logger::AGENCYCOMM)
+      << "Initializing agency failed. We'll try again soon";
+    // We should really have exclusive access, here, this is strange!
+    std::this_thread::sleep_for(std::chrono::seconds(1));
 
     AgencyCommResult result = getValues("InitDone");
```

Comment:
Why this check here? It is checked in the next loop anyway, as far as I see.

Reply:
left over wan't just done yet